libs/ktem/ktem/reasoning/simple.py

```diff
@@ -680,28 +738,90 @@ def stream(  # type: ignore
             history=history,
             evidence=evidence,
             evidence_mode=evidence_mode,
+            sub_question_answer_str=sub_question_answer_str,
             conv_id=conv_id,
             **kwargs,
         )
 
+        if self.use_rewrite:
+            yield Document(
+                channel="info",
+                content=(
+                    "<b>Rewritten pipeline</b><br>"
+                    f"{rewrite_pipeline.__class__.__name__}<br>"
+                    f"<b>Rewritten question</b><br>{message}<br>"
+                ),
+            )
+
+        yield Document(channel="info", content="<b>Evidence</b><br>")
         # show the evidence
         with_citation, without_citation = self.prepare_citations(answer, docs)
         if not with_citation and not without_citation:
-            yield Document(channel="info", content="No evidence found.\n")
+            yield Document(channel="info", content="No evidence found.<br>")
         else:
             yield Document(channel="info", content=None)
             for _ in with_citation:
                 yield _
             if without_citation:
                 yield Document(
                     channel="info",
-                    content="Retrieved segments without matching evidence:\n",
+                    content="Retrieved segments without matching evidence:<br>",
                 )
                 for _ in without_citation:
                     yield _
 
         return answer
 
+    def answer_sub_questions(
+        self, messages: list, conv_id: str, history: list, **kwargs
+    ):
```

Comment:
Please add a docstring and type hint. Reading this without any context, I can't quite figure out what this step is for and why do we need it.